upsint/utils.py

```diff
@@ -111,10 +111,27 @@ def get_remote_url(remote):
     return url.decode("utf-8").strip()
 
 
-def prompt_for_pr_content(commit_msgs):
+def assemble_pr_template(commit_msgs: str, project_pr_template: Optional[str]) -> str:
+    """
+    Prepare text for the prompt to fill in details when creating a new PR.
+    """
+    # first line is always the PR title
+    template = "Title of this PR\n\n"
+    # here we just either show the project's template or just a list of commit messages
+    if project_pr_template:
+        template = f"{template}{project_pr_template}\n{commit_msgs}"
+    else:
+        template = f"{template}PR body:\n{commit_msgs}"
+    return template
+
+
+def prompt_for_pr_content(template: str):
+    """
+    Create a temporary file and feed it to $EDITOR so the user defines content
+    for the PR, mainly title and body.
+    """
     t = tempfile.NamedTemporaryFile(delete=False, prefix="gh.")
```

Comment:
Have a look at: https://click.palletsprojects.com/en/8.0.x/utils/#launching-editors

You don't even need this :)

Reply:
wow, that's sick! I'm glad I know how it works in the background, though you're right we can replace